### PR TITLE
[Doc] Type Widening documentation

### PR DESCRIPTION
**Pull request by johanl-db**
## Description
Add a documentation page for the type widening table feature, in preview in Delta. 3.2

## How was this patch tested?
N/A

### CODE REVIEW

**Review by tomvanbussel — COMMENTED** [review]
Nice! The documentation is very clear. I added a few suggestions to clarify a few things and to integrate this page better in the rest of the docs.

File: docs/source/delta-type-widening.md
[the remark this answers is not in the document]
Comment (johanl-db) [reply docs/source/delta-type-widening.md:1]: Done

File: docs/source/delta-type-widening.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+---
+description: Learn about type widening in Delta.
+---
+
+# Delta type widening
+
+.. note:: This feature is available in preview in <Delta> 3.2.
+
+The type widening feature allows changing the type of columns in a Delta table to a wider type without the need to rewrite the underlying Parquet files.
```
[the remark this answers is not in the document]
Comment (johanl-db) [reply docs/source/delta-type-widening.md]: Updated

File: docs/source/delta-type-widening.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  ALTER TABLE <table_name> SET TBLPROPERTIES ('delta.enableTypeWidening' = 'true')
+  ```
+
+Alternatively, you can enable type widening during table creation:
+
+  ```sql
+  CREATE TABLE T(c1 INT) USING DELTA TBLPROPERTIES('delta.enableTypeWidening' = 'true')
+  ```
+
+Disabling type widening prevents future type changes from being applied to the table. It doesn't affect type changes previously applied and in particular, it doesn't allow clients that don't support the type widening table feature to read and write to the table.
```
[the remark this answers is not in the document]
Comment (johanl-db) [reply docs/source/delta-type-widening.md]: Added an example for disabling the feature

File: docs/source/delta-type-widening.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+To remove the type widening table feature from the table and allow other clients that don't support this feature to read and write to the table, see [_](#removing-the-type-widening-table-feature).
```
[the remark this answers is not in the document]
Comment (johanl-db) [reply docs/source/delta-type-widening.md:45]: 👍

File: docs/source/delta-type-widening.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+```sql
+ALTER TABLE <table_name> CHANGE COLUMN <col_name> TYPE <new_type>
+```
+
+The table schema is updated without rewriting the underlying Parquet files.
+
+## Type changes with automatic schema evolution
+Type changes are applied automatically during ingestion with INSERT and MERGE INTO commands when:
+- Type widening is enabled on the target table.
+- The command runs with [automatic schema evolution](delta-update.md#merge-schema-evolution) enabled.
```
[the remark this answers is not in the document]
Comment (johanl-db) [reply docs/source/delta-type-widening.md:60]: I've left that out for now, given that the feature is still in preview I don't know if we should cross link to it from these more generic pages just yet

File: docs/source/delta-type-widening.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+## Removing the type widening table feature
+
+The type widening feature can be removed from a Delta table using the `DROP FEATURE` command:
+
+```sql
+ ALTER TABLE <table-name> DROP FEATURE 'typeWidening-preview' [TRUNCATE HISTORY]
+```
+
+See [_](delta-drop-feature.md) for more information on dropping Delta table features.
```
[the remark this answers is not in the document]
Comment (johanl-db) [reply docs/source/delta-type-widening.md:74]: 👍

File: docs/source/delta-type-widening.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
+---
+description: Learn about type widening in Delta.
+---
+
+# Delta type widening
+
+.. note:: This feature is available in preview in <Delta> 3.2.
+
+The type widening feature allows changing the type of columns in a Delta table to a wider type. This enables manual type changes using the `ALTER TABLE CHANGE COLUMN` command and automatic type migration with schema evolution in MERGE INTO and INSERT commands.
```
Comment (tomvanbussel) [inline docs/source/delta-type-widening.md:9]: Nit: "`ALTER TABLE CHANGE COLUMN`" is quoted, but "MERGE INTO" and "INSERT" are not.

**Review by tomvanbussel — APPROVED** [review]
One small nit remaining, LGTM otherwise

**Pull request merged**